library/induction.red

```diff
@@ -1,85 +1,81 @@
-import bool
 import nat
 import path
 import unit
 
-def transport (A : type) (M N : A) (p : path A M N) (C : A → type) : C M → C N = 
-	λ cm → coe 0 1 cm in λ i →  C (p i)
+def transport (A : type) (M N : A) (p : path A M N) (C : A → type) : C M → C N =
+  λ cm → coe 0 1 cm in λ i → C (p i)
 
-def le : nat →  nat →  type = 
-	elim [
-		|	zero →  λ _ →  unit
-		| suc (m' → f) → 
-		elim [
-		| zero →  void
-		| suc n' →  f n'
-		]
-	]
+def le : nat → nat → type =
+  elim [
+  | zero → λ _ → unit
+  | suc (m' → f) →
+    elim [
+    | zero → void
+    | suc n' → f n'
+    ]
+  ]
 
-def le/suc (n: nat) : (m : nat) →  le n m → le (suc n) (suc m) =
-	elim n [
-	| zero →  \ _ _  →  triv
-	| suc n' →  \ m' l → l 
-	]
+def le/suc (n: nat) : (m : nat) → le n m → le (suc n) (suc m) =
+  elim n [
+  | zero → λ _ _ → triv
+  | suc n' → λ m' l → l
+  ]
 
-def eq/implies/le (n : nat) : le n n = 
-	elim n [
-		| zero →  triv
-		| suc (n' →  f) →  f
-	]
+def eq/implies/le (n : nat) : le n n =
+  elim n [
+  | zero → triv
+  | suc (n' → f) → f
+  ]
 
-def le/zero/implies/zero (n : nat) : (le n zero) →  path nat zero n = 
-	elim n [
-		| zero →  \ _ →  refl
-		| suc n' →  \ p →  elim p []
-	]
+def le/zero/implies/zero (n : nat) : (le n zero) → path nat zero n =
+  elim n [
+  | zero → λ _ → refl
+  | suc n' → λ p → elim p []
+  ]
 
-def inc : nat → nat = (\ i →  suc i)
+def le/case (m : nat) : (n : nat) → (le n (suc m)) → or (path nat n (suc m)) (le n m) =
```

Comment:
A slicker way to write this is to put `(m : nat)` after the colon, and then change the `elim m` to `elim`.